bin/impl/util.sh

```diff
@@ -16,15 +16,15 @@
 
 function verify_exist_hash() {
   tarball=$1
-  expected_md5=$2
-  actual_md5=$($MD5 "$DOWNLOADS/$tarball" | awk '{print $1}')
+  expected_hash=$2
+  actual_hash=$($HASH_CMD "$DOWNLOADS/$tarball" | awk '{print $1}')
 
   if [[ ! -f "$DOWNLOADS/$tarball" ]]; then
     echo "The tarball $tarball does not exists in downloads/"
     exit 1
   fi
-  if [[ "$actual_md5" != "$expected_md5" ]]; then
-    echo "The MD5 checksum ($actual_md5) of $tarball does not match the expected checksum ($expected_md5)"
+  if [[ "$actual_hash" != "$expected_hash" ]]; then
```

Comment:
there is another place in the code where hashes are compared and it also checks length... this is a nice sanity check to defend against odd cases like comparing  two empty strings.....